source/DmapSupport.cpp

```diff
@@ -327,44 +345,69 @@ HRESULT OpenControllerDevice(PWCHAR deviceName, HANDLE & handle, DWORD shareMode
 #endif // WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
 #if !WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)  // If building a UWP app
+
 /**
-Acquire an exclusive access lock on a controller.
+Send an IO control code to the device driver
 \param[in] handle Handle opened to the device to be locked.
 \return HRESULT success or error code.
 */
-HRESULT GetControllerLock(HANDLE & handle)
+HRESULT SendIoControl(HANDLE & handle, IOControlCode^ iOControlCode)
 {
     HRESULT hr = S_OK;
     CustomDevice^ device;
-    static IOControlCode^ LockCode = ref new IOControlCode(0x423, 0x103, IOControlAccessMode::Any, IOControlBufferingMethod::Neither);
 
     if ((handle < &g_devices[0]) || (handle >= &g_devices[MAX_OPEN_DEVICES]))
     {
-        hr = DMAP_E_INVALID_LOCK_HANDLE_SPECIFIED;
+        return DMAP_E_INVALID_LOCK_HANDLE_SPECIFIED;
```

Comment:
The routine is only used for lock/unlock operations to send IOCTL to controller. So, I've renamed it to SendIoControlCodeToController()